src/executables/util_sa.c

```diff
@@ -3933,49 +3929,86 @@ gen_tz (UTIL_FUNCTION_ARG * arg)
       input_path = inputpath_local;
     }
 
-  /* error message log file */
-  snprintf (er_msg_file, sizeof (er_msg_file) - 1, "%s_%s.err", db_name, arg->command_name);
-  er_init (er_msg_file, ER_NEVER_EXIT);
-  if (tz_gen_type == TZ_GEN_TYPE_EXTEND)
-    {
-      AU_DISABLE_PASSWORDS ();
-      db_set_client_type (DB_CLIENT_TYPE_ADMIN_UTILITY);
-      db_login ("DBA", NULL);
-      if (db_restart (arg->command_name, TRUE, db_name) != NO_ERROR)
-	{
-	  exit_status = EXIT_FAILURE;
-	  goto exit;
-	}
-    }
-
   memset (checksum, 0, sizeof (checksum));
   if (timezone_compile_data (input_path, tz_gen_type, checksum) != NO_ERROR)
     {
       exit_status = EXIT_FAILURE;
       goto exit;
     }
-  if (checksum[0] != '\0')
+
+  if (db_name != NULL)
     {
-      if (put_timezone_checksum (checksum) != NO_ERROR)
+      /* error message log file */
+      snprintf (er_msg_file, sizeof (er_msg_file) - 1, "%s_%s.err", db_name, arg->command_name);
+      er_init (er_msg_file, ER_NEVER_EXIT);
+    }
+  if (tz_gen_type == TZ_GEN_TYPE_EXTEND && checksum[0] != '\0')
+    {
+      AU_DISABLE_PASSWORDS ();
+      db_set_client_type (DB_CLIENT_TYPE_ADMIN_UTILITY);
+      db_login ("DBA", NULL);
+
+      if (db_name != NULL)
+	{
+	  dir = (DB_INFO *) calloc (1, sizeof (DB_INFO));
+	  if (dir == NULL)
+	    {
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (DB_INFO));
+	      exit_status = EXIT_FAILURE;
+	      goto exit;
+	    }
+	  dir->name = (char *) calloc (strlen (db_name) + 1, sizeof (char));
+	  if (dir->name == NULL)
+	    {
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (DB_INFO));
```

Comment:
Dear Mr Oh,

The dir variable is of DB_INFO type so the allocation with sizeof(DB_INFO) is correct, strlen(db_name) + 1 is used when I allocate memory for the db_name variable.

                                                   Best regards,
                                                             Antonio